### PR TITLE
Feat/initial pr decoration

### PR DESCRIPTION
**Pull request by joachimprinzbach**

### CODE REVIEW

**Comment by incubator-tekton** [comment]
This PR is managed by **Argonaut** 

 You can use the command `/ping` as pull request command to test the interaction in a comment. 

 You can use the command `/deploy {image_tag}` to deploy an image into this preview environment. 

 You can use the command `/promote {image_tag}` to deploy an image to production.

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
Successfully deployed version latest

**Comment by incubator-tekton** [comment]
Successfully deployed version e latest

**Comment by incubator-tekton** [comment]
Successfully deployed version latest

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by joachimprinzbach** [comment]
/promote feat-initial_pr_decoration-fa3b4db1696615d16c849f1fc7c78d1655046aa3

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-initial_pr_decoration-fa3b4db1696615d16c849f1fc7c78d1655046aa3

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-initial_pr_decoration-79ff0691a080835d438a0c460d1fb21820016c84

**Comment by incubator-tekton** [comment]
Successfully deployed version master-1c46631de781d99376232c3991b9a881c5bad1c9

**Comment by incubator-tekton** [comment]
Successfully deployed version master-5327885961ab429fdc9433b81584ca10d2a2e22a

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-initial_pr_decoration-5327885961ab429fdc9433b81584ca10d2a2e22a

**Comment by incubator-tekton** [comment]
Successfully deployed version master-5327885961ab429fdc9433b81584ca10d2a2e22a

**Comment by incubator-tekton** [comment]
Successfully deployed version master-1d76e17fdcd2315734950f4359ea95bd2989ee18

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-initial_pr_decoration-1d76e17fdcd2315734950f4359ea95bd2989ee18

**Comment by incubator-tekton** [comment]
Successfully deployed version master-1d76e17fdcd2315734950f4359ea95bd2989ee18

**Comment by incubator-tekton** [comment]
pong!

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-initial_pr_decoration-1d76e17fdcd2315734950f4359ea95bd2989ee18

**Comment by incubator-tekton** [comment]
Successfully deployed version feat-initial_pr_decoration-1d76e17fdcd2315734950f4359ea95bd2989ee18

**Comment by incubator-tekton** [comment]
pong!

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 1]
```diff
@@ -66,41 +67,43 @@ public void handleGitHubWebhookEvent(@RequestBody String data, @RequestHeader(GI
...
-                case "opened": {
-                    LOGGER.info("PR OPENED Event");
-                    break;
-                }
-                default: {
-                    LOGGER.info("Unhandled GitHub Pull Request Event Action: {}", pullRequest.getAction());
-                }
+    private void handlePullRequestEvent(GHEventPayload.PullRequest ghPullRequest) {
+        switch (ghPullRequest.getAction()) {
+            case "closed": {
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java:72]: Would constants or enum for "closed" and "opened" be better?

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 2]
```diff
@@ -66,41 +67,43 @@ public void handleGitHubWebhookEvent(@RequestBody String data, @RequestHeader(GI
...
 
     private void handlePushEvent(GHEventPayload.Push push) {
         LOGGER.info("REPO PUSH Event");
     }
 
-    private void handleIssueCommentEvent(GHEventPayload.IssueComment issueComment) {
+    private void handleIssueCommentEvent(GHEventPayload.IssueComment issueComment) throws IOException {
         switch (issueComment.getAction()) {
             case "created": {
                 String commentText = issueComment.getComment().getBody();
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java:95]: Violation of Law of Demeter, maybe a method getCommentBody() can solve it.

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 3]
```diff
@@ -66,41 +67,43 @@ public void handleGitHubWebhookEvent(@RequestBody String data, @RequestHeader(GI
...
                 GHRepository repository = issueComment.getRepository();
-                PullRequest pullRequest = new PullRequest(issueComment.getIssue().getNumber(), repository.getOwnerName(), repository.getName());
-                if (commentText.startsWith("/ping")) {
+                GHPullRequest ghPullRequest = repository.getPullRequest(issueComment.getIssue().getNumber());
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Violation of Law of Demeter, maybe a method getIssueNumber() can solve it.

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 4]
```diff
@@ -66,41 +67,43 @@ public void handleGitHubWebhookEvent(@RequestBody String data, @RequestHeader(GI
...
+                PullRequest pullRequest = createPullRequest(issueComment.getIssue().getNumber(), repository, ghPullRequest);
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Duplicate Number getter, maybe use a variable

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 5]
```diff
@@ -66,41 +67,43 @@ public void handleGitHubWebhookEvent(@RequestBody String data, @RequestHeader(GI
...
+                if ("/ping".startsWith(commentText)) {
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java:98]: Are more commands planned? If yes you should think about another pattern for better adding new ones. Maybe also use constants for all the key words.

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java [hunk 6]
```diff
@@ -114,7 +117,8 @@ private void handleIssueCommentEvent(GHEventPayload.IssueComment issueComment) {
         }
     }
 
-    private String getActionElementValue(JsonObject jsonObject) {
-        return jsonObject.get("action").getAsString();
+    private PullRequest createPullRequest(int id, GHRepository repository, GHPullRequest pullRequest) {
+        return new PullRequest(id, repository.getSvnUrl(), repository.getOwnerName(), repository.getName(), pullRequest.getHead().getRef(), pullRequest.getHead().getSha());
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: There are a lot of parameters, maybe you can offer an additional constructor who has a repository as parameter, so you can reduce the parameters count.

Comment (leubedane) [reply src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Another Question: What is the SvnUrl ? SourceVersioning ...?

File: src/main/java/com/baloise/incubator/argonaut/infrastructure/git/GitDeployPullRequestService.java [hunk 7]
```diff
@@ -36,20 +37,22 @@
...
-        } catch (IOException e) {
-            LOGGER.error("Error getting PR HEAD REF: ", e);
-        }
-        LOGGER.info("Deploying url: {}, pullRequest: {}, newImageTag: {}", deploymentRepoUrl, pullRequest, newImageTag);
-        String sanitizedBranchName = branchName.replace("/", "-");
+    public void promoteToProd(PullRequest pullRequest) {
+        deployBranch(pullRequest, true);
+    }
+
+    private void deployBranch(PullRequest pullRequest, boolean isProd) {
```
Comment (leubedane) [inline src/main/java/com/baloise/incubator/argonaut/infrastructure/git/GitDeployPullRequestService.java:50]: I dont know if the flag isProd has the right name. Maybe you can invert the boolean and name it useBranchName or branchSpecificEnvironment, . Or let it and name it useMasterBranch.  (or am I thinking to complicated ;-))

**Review by leubedane — APPROVED** [review]

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java (hunk 2 above)
Comment (joachimprinzbach) [reply src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java:95]: Das Objekt ist aus einem SDK, ned vo mir ;-)

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java (hunk 3 above)
Comment (joachimprinzbach) [reply src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Objekt is aus einem SDK

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java (hunk 4 above)
Comment (joachimprinzbach) [reply src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Fixed

File: src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java (hunk 6 above)
Comment (joachimprinzbach) [reply src/main/java/com/baloise/incubator/argonaut/application/github/GitHubWebhookRestController.java]: Thats how it is called in GitHub. 
Probably I need to come up with a property name thats fitting better. Its the clone URL, but without any git stuff. 
e.g. https://github.com/org/repo

File: src/main/java/com/baloise/incubator/argonaut/infrastructure/git/GitDeployPullRequestService.java (hunk 7 above)
Comment (joachimprinzbach) [reply src/main/java/com/baloise/incubator/argonaut/infrastructure/git/GitDeployPullRequestService.java:50]: Its basically a quick hack for the moment. You do not want to deploy to prod from a feature branch i guess. It makes things easier to test for the moment.

**Pull request merged**